src/websocket/broadcast/gameBroadcast.ts

```diff
@@ -228,24 +230,58 @@ export const broadcastGameStart = async (
   await Promise.all(promises);
 };
 
+/**
+ * Broadcast DRAW_ROUND to users when the wall runs out of tile.
+ * @param {WebSocketClient} ws a WebSocketClient instance
+ * @param {string} gameId Game Id
+ * @param {string} connectionId Connection Id
+ * @param {string[]} connectionIds All connection Ids in the game
+ */
+export const broadcastDrawRound = async (
+  ws: WebSocketClient,
+  gameId: string,
+  connectionId: string,
+  connectionIds: string[],
+): Promise<void> => {
+  const wsResponse = createDrawRoundResponse({ gameId, connectionId });
+  await Promise.all(connectionIds.map((cid) => ws.send(wsResponse, cid)));
+};
+
 /**
  * Broadcast a tile string to a user in the game.
  * @param {WebSocketClient} ws a WebSocketClient instance
- * @param {string} gameId game Id
- * @param {string} connectionId connection Id
+ * @param {string} gameId Game Id
+ * @param {string} connectionId Connection Id
  */
 export const broadcastDrawTileToUser = async (
   ws: WebSocketClient,
   gameId: string,
   connectionId: string,
-): Promise<string> => {
+): Promise<void> => {
   const tileDrawn = await drawTile(gameId);
-  console.log('tile drawn:', tileDrawn);
 
-  const wsResponse = createDrawTileResponse({ tile: tileDrawn });
-  await ws.send(wsResponse, connectionId);
+  if (!tileDrawn || tileDrawn === '') {
+    // Double check to make sure tile index reach 143
+    const currentIndex = (await getCurrentTileIndex(gameId)) as number;
+    if (currentIndex !== Wall.DEFAULT_WALL_LENGTH - 1) {
+      throw Error('broadcastDrawTileToUser: Draw empty tile while currentIndex did not reach 144');
+    }
+
+    // Users must exist in the game
+    const users = await getUsersInGame(gameId);
+    if (!users) throw Error('broadcastDrawTileToUser: No user found in game, failed to broadcast DRAW_ROUND');
+
+    const connectionIds = getConnectionIdsFromUsers(users);
+    await broadcastDrawRound(ws, gameId, connectionId, connectionIds);
+    return;
+  }
 
-  return tileDrawn;
+  console.log(`Tile ${tileDrawn} was drawn by ${connectionId}`);
+  const wsResponse = createDrawTileResponse({
+    tile: tileDrawn,
+    currentIndex: (await getCurrentTileIndex(gameId)) as number,
+  });
```

Comment:
I see you're passing in currentIndex as well. I will update front-end for this change

Reply:
kk